pandas/io/gbq.py

```diff
@@ -148,15 +168,63 @@ def get_credentials(self):
 
         return credentials
 
+    def get_service_account_credentials(self):
+        from oauth2client.client import SignedJwtAssertionCredentials
+        from os.path import isfile
+
+        try:
+            if isfile(self.private_key):
+                with open(self.private_key) as f:
+                    json_key = json.loads(f.read())
+            else:
+                # ugly hack: 'private_key' field has new lines inside, they break json parser, but we need to preserve them
+                json_key = json.loads(self.private_key.replace('\n', '   '))
+                json_key['private_key'] = json_key['private_key'].replace('   ', '\n')
+
+            if compat.PY3:
+                json_key['private_key'] = bytes(json_key['private_key'], 'UTF-8')
+
+            return SignedJwtAssertionCredentials(
+                json_key['client_email'],
+                json_key['private_key'],
+                self.scope,
+            )
+        except (KeyError, ValueError, TypeError, AttributeError):
+            raise InvalidPrivateKeyFormat("Service account private key should be valid JSON (file path or string contents) "
+                "with at least two keys: 'client_email' and 'private_key'. Can be obtained from google developers console. ")
```

Comment:
Can we make a small change in the error message to indicate that file may be missing?

```
raise InvalidPrivateKeyFormat("Private key is missing or invalid. Service account private key should be valid JSON (file path or string contents) with at least two keys: 'client_email' and 'private_key'. Can be obtained from google developers console. ")
```